Adds Bitbucket cloud token support (#30774)
backstage/backstage: .changeset/cute-taxis-double.md

---

> '@backstage/backend-defaults': minor
> '@backstage/integration': minor

Reviewer: ```suggestion
'@backstage/backend-defaults': patch
'@backstage/integration': patch
```

There's no breaking changes in this, you are just adding the support for `token` as well. If this is `patch` then the Maintainers would be open to including this in a patch of the current `1.43.x` release.

---
'@backstage/backend-defaults': minor
'@backstage/integration': minor
---

> **BREAKING** Bitbucket cloud now supports API token as App password will be deprecated soon.

Reviewer 2: ```suggestion
**Attention:** Support for Bitbucket Cloud's API token was added. App password is deprecated (no new creation from September 9, 2025) and will be removed on June 9, 2026.
```

So far, it is still a minor change, not a major/breaking change due to the provided backwards compatibility.

```yaml
integrations:
  bitbucketCloud:

>     - host: bitbucket.org
>       # Option 1 (recommended): API Token
>       token: ${BITBUCKET_API_TOKEN}

Reviewer 2: ```suggestion
    # Option 1 (recommended): API Token
    - token: ${BITBUCKET_API_TOKEN}
```
`host` is obsolete/static. It cannot be another value. There is no reason for the user to be able to set a value.

---
'@backstage/backend-defaults': minor
'@backstage/integration': minor
---

**Attention:** Support for Bitbucket Cloud's API token was added. App password is deprecated (no new creation from September 9, 2025) and will be removed on June 9, 2026.

API token (recommended):

```yaml
integrations:
  bitbucketCloud:

>     - username: user@domain.com

Reviewer: Can you clarify what you mean?

>     - username: user@domain.com

Reviewer 3: The file currently contains commented example configs for the integration using `appPassword`. See [here](https://github.com/backstage/backstage/blob/fd370cab1d923e31afd4948d961b9e297dc81264/app-config.yaml#L109). This should be updated to use `token` instead.

>     - username: user@domain.com

Author: I think it's related to the example usage [here](https://github.com/backstage/backstage/blob/fd370cab1d923e31afd4948d961b9e297dc81264/app-config.yaml#L108)

>     - username: user@domain.com

Reviewer: That file is simply for the example app here for local development when contributing, I don't think that's needed, to be honest all of those examples would be better as links to the actual docs.

      token: my-token
```

App password (deprecated):

```yaml
integrations:
  bitbucketCloud:
    - username: user
      appPassword: my-secret

> ```

Reviewer: Shoot, forgot to mention we should also update the docs here as well: https://github.com/backstage/backstage/blob/master/docs/integrations/bitbucketCloud/locations.md
